bank/persistence/account_operations_test.go

```diff
@@ -0,0 +1,103 @@
+package persistence_test
+
+import (
+	"github.com/iliyanmotovski/bankv1/bank/domain"
+	"github.com/iliyanmotovski/bankv1/bank/persistence"
+	"github.com/iliyanmotovski/bankv1/bank/persistence/db_config"
+	"gopkg.in/mgo.v2/bson"
+	"testing"
+)
+
+func TestItWritesAndReadsAccountData(t *testing.T) {
+	db := db_config.NewDatabase()
+	defer db.Close()
+	defer db.Clean()
+
+	accountStore := persistence.NewAccountStore(*db.Database.Session, db.Database.Name)
+
+	testAccount := domain.Account{UserID: "1234", Currency: "testC", Amount: 10, Type: "VISA"}
+
+	accountID, err := accountStore.InsertAccount(testAccount.UserID, testAccount)
+	if err != nil {
+		t.Error(err.Error())
+	}
+	testAccount.AccountID = accountID
+
+	a, err := accountStore.GetAccounts(testAccount.UserID)
+	if err != nil {
+		t.Error(err.Error())
+	}
+	account := *a
+
+	if account[0] != testAccount {
+		t.Errorf("expected account: %v got: %v", testAccount, account[0])
+	}
+}
+
+func TestItUpdatesAndDeletesAccountData(t *testing.T) {
+	db := db_config.NewDatabase()
+	defer db.Close()
+	defer db.Clean()
+
+	accountStore := persistence.NewAccountStore(*db.Database.Session, db.Database.Name)
+
+	db.Database.C("accounts").Insert(&domain.Account{UserID: "1234", AccountID: "4321", Currency: "testC", Amount: 10})
+
+	testAccountDeposit := domain.Account{UserID: "1234", Currency: "testC", Amount: 5}
+	testAccountWithdraw := domain.Account{UserID: "1234", Currency: "testC", Amount: 2}
+	expected := domain.Account{UserID: "1234", AccountID: "4321", Currency: "testC", Amount: 13}
+
+	err := accountStore.Deposit(testAccountDeposit)
+	if err != nil {
+		t.Error(err.Error())
+	}
+
+	err = accountStore.Withdraw(testAccountWithdraw)
+	if err != nil {
+		t.Error(err.Error())
+	}
+
+	account := []domain.Account{}
+	db.Database.C("accounts").Find(bson.M{"userid": "1234", "currency": "testC"}).All(&account)
+
+	if account[0] != expected {
+		t.Errorf("expected account: %v got: %v", expected, account[0])
+	}
+
+	err = accountStore.DeleteAccount("1234", "4321")
+	if err != nil {
+		t.Error(err.Error())
+	}
+
+	db.Database.C("accounts").Find(bson.M{"userid": "1234", "currency": "testC"}).All(&account)
```

Comment:
When you retrieve list of all accounts it should be empty. Use it instead of using of the low level mgo API in test. This is smelly as it duplicates names of fields and collection between test and production code.